denonavr/input.py

```diff
@@ -155,25 +160,211 @@ def setup(self) -> None:
             return
 
-        await self.async_update_media_state()
+        if parameter != POWER_ON:
+            self._stop_media_update()
+            self._state = STATE_OFF
+            self._title = None
+            self._artist = None
+            self._album = None
+            self._band = None
+            self._frequency = None
+            self._station = None
+            self._image_url = None
+        elif self._input_func in self._netaudio_func_list:
+            self._stop_media_update()
+            self._schedule_netaudio_update()
+        elif self._input_func in ["Tuner", "TUNER"]:
+            self._stop_media_update()
+            self._schedule_tuner_update()
+        elif self._input_func in ["HD Radio", "HDRADIO"]:
+            self._stop_media_update()
+            self._schedule_hdtuner_update()
+
+    def _stop_media_update(self) -> None:
+        """Stop the media update task."""
+        if self._media_update_handle is not None:
+            self._media_update_handle.cancel()
+            self._media_update_handle = None
+
+    def _schedule_netaudio_update(self) -> None:
+        """Schedule a netaudio update task."""
+        self._state = STATE_PLAYING
+        loop = asyncio.get_event_loop()
+        self._media_update_handle = loop.call_later(
+            _MEDIA_UPDATE_INTERVAL,
+            self._update_netaudio
+        )
+
+    def _update_netaudio(self) -> None:
+        """Update netaudio information."""
+        self._device.telnet_api.send_command("NSE")
+        self._schedule_netaudio_update()
+
+    async def _async_netaudio_callback(
+            self, zone: str, event: str, parameter: str) -> None:
+        """Handle a netaudio update event."""
+        if self._device.power != POWER_ON:
+            return
+        if self._input_func not in self._netaudio_func_list:
+            return
+
+        if parameter.startswith("E1"):
+            self._title = parameter[2:].strip()
+        elif parameter.startswith("E2"):
+            self._artist = parameter[2:].strip()
+        elif parameter.startswith("E4"):
+            self._album = parameter[2:].strip()
+        self._band = None
+        self._frequency = None
+        self._station = None
+
+        # Refresh cover with a hash for media URL when track is changing
+        self._image_url = (ALBUM_COVERS_URL.format(
+            host=self._device.api.host, port=self._device.api.port,
+            hash=hash((self._title, self._artist, self._album))))
+        await self._async_test_image_accessable()
+
+    def _schedule_tuner_update(self) -> None:
+        """Schedule a tuner update task."""
+        self._state = STATE_PLAYING
+        loop = asyncio.get_event_loop()
+        self._media_update_handle = loop.call_later(
+            _MEDIA_UPDATE_INTERVAL,
+            self._update_tuner
+        )
+
+    def _update_tuner(self) -> None:
+        """Update tuner information."""
+        self._device.telnet_api.send_command("TFAN?")
```

Comment:
If we need to send multiple commands in more places it might make sense to make send_command send_commands and accept an intervals